src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -659,21 +667,19 @@ Optional<byte[]> getBytesFromContext(ParseContext context, int dimension) throws
         return Optional.of(array);
     }
 
-    Optional<float[]> getFloatsFromContext(ParseContext context, int dimension) throws IOException {
+    Optional<float[]> getFloatsFromContext(ParseContext context, int dimension, MethodComponentContext methodComponentContext)
+        throws IOException {
         context.path().add(simpleName());
 
         // Returns an optional array of float values where each value in the vector is parsed as a float and validated
         // if it is a finite number and within the fp16 range of [-65504 to 65504] by default if Faiss encoder is SQ and type is 'fp16'.
         // If the encoder parameter, "clip" is set to True, if the vector value is outside the FP16 range then it will be
         // clipped to FP16 range.
-        boolean isFaissSQfp16Flag = isFaissSQfp16(fieldType().getKnnMethodContext());
+        boolean isFaissSQfp16Flag = isFaissSQfp16(methodComponentContext);
         boolean clipVectorValueToFP16RangeFlag = false;
         if (isFaissSQfp16Flag) {
             clipVectorValueToFP16RangeFlag = isFaissSQClipToFP16RangeEnabled(
-                (MethodComponentContext) fieldType().getKnnMethodContext()
-                    .getMethodComponentContext()
-                    .getParameters()
-                    .get(METHOD_ENCODER_PARAMETER)
+                (MethodComponentContext) methodComponentContext.getParameters().get(METHOD_ENCODER_PARAMETER)
```

Comment:
Can we cast safely? Check type first

Reply:
Here, we are casting it only when the if condition is true which means it is valid as we have already verified above. So, I feel we can skip the validation here.